src/libraries/System.Security.Cryptography/tests/ShakeTestDriver.cs

```diff
@@ -0,0 +1,448 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Security.Cryptography.Tests
+{
+    public interface IShakeTrait<TShake> where TShake : IDisposable, new()
+    {
+        static abstract TShake Create();
```

Comment:
Do you need a Create() given that `TShake : new()`?  The tests seemed to be generally using `new TShake()` instead of `TShakeTrait.Create()`.

Reply:
Yeah. That's explicitly done in this commit: https://github.com/dotnet/runtime/pull/87099/commits/dea682d33e48cbc2992c710bbc9eb861543ed0de

"Why?"

Because we want to test that the constructor throws PNSE on unsupported platforms. But `new T()` for whatever reason [wraps the ctor exception in System.Reflection.TargetInvocationException](https://sharplab.io/#v2:C4LgTgrgdgNAJiA1AHwAICYAMBYAUBgRj2NygFMB3AAgAUB7YAHgDEw6BzAPgAoBKAbhIZaDRgBVOVCgAsyYMlTEhyFPlQDeeKtqqoAzCOBqAvJJWK+g3AF8h6Kqw4atO/Q7bsTk4NLbVzNAA2AIbAAGZ0YAC2AHIMAMoQAA5JkcBkcACiAB4AxmRJwACWdFCWeNZAA=).

Testing that it threw TargetInvocationException didn't "feel" like the right test to me.